packages/@aws-cdk/aws-s3-deployment/lib/source.ts

```diff
@@ -54,9 +53,14 @@ export class Source {
    * @param path The path to a local .zip file or a directory
    */
   public static asset(path: string): ISource {
+    const pathHash = crypto.createHash('md5')
```

Comment:
i don't think a hash is needed. construct ID is already managed and hashed as needed. you can literally put any string there. I would use `Asset(${path})` as the construct ID.